services/director-v2/src/simcore_service_director_v2/api/dependencies/database.py

```diff
@@ -14,26 +14,38 @@ def _get_db_engine(request: Request) -> Engine:
     return request.app.state.engine
 
 
+def _get_base_repository(
+    engine: Engine, repo_type: Type[BaseRepository]
+) -> BaseRepository:
+    # NOTE: 2 different ideas were tried here with not so good
+    # 1st one was acquiring a connection per repository which lead to the following issue https://github.com/ITISFoundation/osparc-simcore/pull/1966
+    # 2nd one was acquiring a connection per request which works but blocks the director-v2 responsiveness once
+    # the max amount of connections is reached
+    # now the current solution is to acquire connection when needed.
+
+    available_engines = engine.maxsize - (engine.size - engine.freesize)
+    if available_engines <= 1:
+        logger.warning(
+            "Low pg connections available in pool: pool size=%d, acquired=%d, free=%d, reserved=[%d, %d]",
+            engine.size,
+            engine.size - engine.freesize,
+            engine.freesize,
+            engine.minsize,
+            engine.maxsize,
+        )
+    return repo_type(db_engine=engine)
+
+
 def get_repository(repo_type: Type[BaseRepository]) -> Callable:
     async def _get_repo(
         engine: Engine = Depends(_get_db_engine),
     ) -> AsyncGenerator[BaseRepository, None]:
-        # NOTE: 2 different ideas were tried here with not so good
-        # 1st one was acquiring a connection per repository which lead to the following issue https://github.com/ITISFoundation/osparc-simcore/pull/1966
-        # 2nd one was acquiring a connection per request which works but blocks the director-v2 responsiveness once
-        # the max amount of connections is reached
-        # now the current solution is to acquire connection when needed.
-
-        available_engines = engine.maxsize - (engine.size - engine.freesize)
-        if available_engines <= 1:
-            logger.warning(
-                "Low pg connections available in pool: pool size=%d, acquired=%d, free=%d, reserved=[%d, %d]",
-                engine.size,
-                engine.size - engine.freesize,
-                engine.freesize,
-                engine.minsize,
-                engine.maxsize,
-            )
-        yield repo_type(db_engine=engine)
+        yield _get_base_repository(engine=engine, repo_type=repo_type)
 
     return _get_repo
+
+
+def fetch_repo_no_request(
```

Comment:
not too sure if that is where this should be. it's a dependencies module for the API right?